aries/aries_vcx_core/src/anoncreds/anoncreds.rs

```diff
@@ -0,0 +1,1799 @@
+                    .get_wallet_record_value(wallet, CATEGORY_REV_REG, rev_reg_def_id)
+                    .await?;
+                let rev_reg_info: RevocationRegistryInfo = self
+                    .get_wallet_record_value(wallet, CATEGORY_REV_REG_INFO, rev_reg_def_id)
+                    .await?;
+
+                let rev_reg_def_id = RevocationRegistryDefinitionId::new(rev_reg_def_id).unwrap();
+
+                Some((
+                    rev_reg_def,
+                    rev_reg_def_id,
+                    rev_reg_def_priv,
+                    rev_reg,
+                    rev_reg_info,
+                    tails_dir,
+                ))
+            }
+            (None, None) => None,
+            (tails_dir, rev_reg_def_id) => {
+                warn!(
+                    "Missing revocation config params: tails_dir: {tails_dir:?} - \
+                     {rev_reg_def_id:?}; Issuing non revokable credential"
+                );
+                None
+            }
+        };
+
+        let rev_status_list = match &revocation_config_parts {
+            Some((rev_reg_def, rev_reg_def_id, rev_reg_def_priv, _, _, _)) => {
+                Some(create_revocation_status_list(
+                    &cred_def,
+                    rev_reg_def_id.clone(),
+                    rev_reg_def,
+                    rev_reg_def_priv,
+                    true,
+                    None,
+                )?)
+            }
+            _ => None,
+        };
+
+        let revocation_config = match (&mut revocation_config_parts, &rev_status_list) {
+            (
+                Some((rev_reg_def, _, rev_reg_def_priv, _, rev_reg_info, _)),
+                Some(rev_status_list),
+            ) => {
+                rev_reg_info.curr_id += 1;
+
+                if rev_reg_info.curr_id > rev_reg_def.value.max_cred_num {
+                    return Err(AriesVcxCoreError::from_msg(
+                        AriesVcxCoreErrorKind::ActionNotSupported,
+                        "The revocation registry is full",
+                    ));
+                }
+
+                rev_reg_info.used_ids.insert(rev_reg_info.curr_id);
+
+                let revocation_config = CredentialRevocationConfig {
+                    reg_def: rev_reg_def,
+                    reg_def_private: rev_reg_def_priv,
+                    registry_idx: rev_reg_info.curr_id,
+                    status_list: rev_status_list,
+                };
+
+                Some(revocation_config)
+            }
+            _ => None,
+        };
+
+        let cred = anoncreds::issuer::create_credential(
+            &cred_def,
+            &cred_def_private,
+            &cred_offer,
+            &cred_request,
+            cred_values,
+            revocation_config,
+        )?;
+
+        let rev_reg = cred.rev_reg.as_ref();
+
+        let cred_rev_id =
+            if let (Some(rev_reg_id), Some(rev_reg), Some((_, _, _, _, rev_reg_info, _))) =
```

Comment:
Again, I know this is mostly copy-paste, but I'll take opportunity to comment this. We should only be testing one (preferably the input `rev_reg_id`, as that the initial source of truth as whether we issue revocable or irrevocable credential), not all tree for `Some`, and throw an error else.
If first of the tuple is `Some`, but other are `None`, something must had gone unexpectedly wrong, likely due a bug - so we shouldn't instead silently treat that same way as we treat irrevocable credentials.